### PR TITLE
Fixed partitions reset issue when node gets isolated

### PR DESCRIPTION
**Pull request by goran-ethernal**
Fixed the issue on partitions reset when reverting `PartitionAction`, where a node gets isolated from the rest of the network and stops communicating with other nodes.

When calling Reset on a partition, `PartitionTransport` subsets were set to an empty map, which can lead to isolation of a given node, since `Gossip` will check if a node is connected to another node, and since its subset is empty it will return false and node will get stuck.

This is solved by setting subsets map on `PartitionTransport` to `nil` on `Reset`, and `nil` is handled in `isConnected` method as a valid case which means that given node is connected to the entire network.
When another `PartitionAction` is created it will set it its map to another partition and action can proceed to execute.

**NOTE** For flow map, given behavior will be implemented and pushed when it gets merged with this branch.

### CODE REVIEW

**Review by Stefan-Ethernal — APPROVED** [review]
LGTM, but let's wait for Ferran's approval.

**Review by ferranbt — APPROVED** [review]

**Pull request merged**
